### PR TITLE
IRSR24-28 | @jdwjdwjdw | Add search button to nav

### PR DESCRIPTION
**Pull request by jdwjdwjdw**
# READY FOR REVIEW

# Summary
- [IRSR24-28](https://stanfordits.atlassian.net/browse/IRSR24-28): Search in Navigation
  - Search icon inline with main nav links
  - No search field needed
  - Search icon takes user to the search page

Screenshot 2023-11-01 at 3 14 19 PM

# Review By (Date)
- EOD 11/17

# Urgency
- Normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Review site and confirm that the search icon is visible and matches [figma](https://www.figma.com/file/C28Tjq613J2ZK1OmzXdMkB/%5BIRSR%5D-Project-Starter-File?type=design&node-id=966-13879&mode=design&t=SciWFsUguGvuJjcM-0)/[the ticket](https://stanfordits.atlassian.net/browse/IRSR24-28) specifications (Search icon inline with main nav links, No search field needed, Search icon takes user to the search page).
3. Review code 🏂 

# Associated Issues and/or People
- [IRSR24-28](https://stanfordits.atlassian.net/browse/IRSR24-28): Search in Navigation

[IRSR24-28]: https://stanfordits.atlassian.net/browse/IRSR24-28?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/scss/components/search/_search.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  .su-multi-menu {
+    display: flex;
+  
+    .menu-search-button {
+      position: unset;
+      margin-left: 3.6rem;
+      align-self: center;
+      width: 2.6rem;
```
Comment (rebeccahongsf) [inline src/scss/components/search/_search.scss]: The search icon appears a bit bigger than the neighboring text

image

Would you be able to update the sizing to match the figma mock? 😃 

image

File: src/js/ppo_irsr_subtheme.behaviors.js [hunk 2]
```diff
@@ -2,6 +2,8 @@
   'use strict';
   Drupal.behaviors.ppoIrsrSubtheme = {
     attach: function (context) {
+      const menuSearchButton = $('<a>', { class: 'menu-search-button su-site-search__submit', 'href': '/search', 'aria-label': 'Go to search page' });
+      $('.su-multi-menu', context).append(menuSearchButton);
```
Comment (rebeccahongsf) [inline src/js/ppo_irsr_subtheme.behaviors.js]: Is there a way we can add the search icon here instead of a background image?

**Comment by jdwjdwjdw** [comment]
Thanks @rebeccahongsf! I decided to take your suggestion and update the search icon to using font awesome, as my previous solution didn't display in situations where images are disabled. The sizing should now better match figma, and I updated the color to match figma as well (you'll see that it's using `$su-color-cardinal-red-dark`, so slightly darker than the links). Let me know if you see any further adjustments needed!

File: src/scss/components/search/_search.scss [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        &:active {
+          color: $su-color-black;
+        }
+      }
+    }
+  }
+}
+
+@include grid-media-max('md') {
+  .menu-search-button {
```
Comment (rebeccahongsf) [inline src/scss/components/search/_search.scss]:
```suggestion
  .menu-search-link {
```
Just gotta update this to the correct target class and we should be good to go! 😄

**Review by rebeccahongsf — CHANGES_REQUESTED** [review]
Thank you for implementing the changes @jdwjdwjdw 😃 It looks great to me on desktop!
image

I did notice that the search button appears on mobile
image

Once that's updated, we should be good to go!

**Comment by jdwjdwjdw** [comment]
Good catch @rebeccahongsf! Updated and should be good to go now.

**Review by rebeccahongsf — APPROVED** [review]
Everything looks awesome! Thank you @jdwjdwjdw 🚢

**Pull request merged**